general/session.md

```diff
@@ -0,0 +1,26 @@
+# Session API
+
+## Workflow
+
+### Getting
+
+When getting a value stored in Session, we are deserializing it by removing the `serialize://` string from the data we retrieve. After the data is deserialized we get access to the data requested, stripped of any serialization.
+
+#### Needs clarifying
+
+-   Even though we are using POST we are sending a `'cmd': 'get'`, which might be revealing the underlying architecture of how Liferay handles fetch requests.
+-   Why are we using a POST method in a getter?
```

Comment:
Probably doesn't need to be addressed in this documentation, which is mostly about usage. The point of having a `getSessionValue()` method is to abstract over these details so that the developer doesn't have to worry about them too much.

Talking about `serialize://` is getting into implementation details, but I think it's an acceptable level of detail to go into, especially if we reorder the sections so that `setSessionValue()` comes first and we explain how the prefix gets put in there in the first place.

Reply:
This was mainly for myself, so I can  understand the API better in order to either explain it better before merging, so if you know why we're using `POST` instead of `GET`, and how the whole `'cmd': 'get'` thingy functions it would be nice 😄